### PR TITLE
IGNITE-18206 Sql. Prohibit using 2-phase aggregates with distinct accumulator on map phase

### PR DESCRIPTION
**Pull request by zstan**

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rule/HashAggregateConverterRule.java [hunk 1]
```diff
@@ -74,6 +77,21 @@ protected PhysicalNode convert(RelOptPlanner planner, RelMetadataQuery mq,
         }
     }
 
+    /**
+     * Return {@code true} if observes COUNT and DISTINCT simultaneously in aggregate.
+     *
+     * @param aggCalls Aggregates.
+     * @return {@code true} If found, {@code false} otherwise.
+     */
+    private static boolean countWithDistinctAgg(List<AggregateCall> aggCalls) {
```
Comment (korlov42) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rule/HashAggregateConverterRule.java:86]: what about other aggregate functions?

Comment (zstan) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rule/HashAggregateConverterRule.java]: /me just narrow thinking ) fixed

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/AggregatePlannerTest.java [hunk 2]
```diff
@@ -372,6 +373,63 @@ public void colocated(AggregateAlgorithm algo) throws Exception {
...
+
+    /**
+     * Check that plan does not contain distinct accumulators on map nodes.
+     *
+     * @param sql Request string.
+     * @param publicSchema Schema.
+     * @throws Exception If failed.
+     */
+    private void checkDistinctInMapAggNode(String sql, IgniteSchema publicSchema) throws Exception {
+        String[][] disabledRules = new String[][] {{""}, {"MapReduceHashAggregateConverterRule"},
```
Comment (korlov42) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/AggregatePlannerTest.java:419]: Why do we need an empty string here?

**Review by korlov42 — APPROVED** [review]

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/AggregatePlannerTest.java (hunk 2 above)
Comment (zstan) [reply modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/AggregatePlannerTest.java:419]: discussed in messenger, will be change a bit.

**Pull request closed without merging**